Sources/Noora/Components/SingleChoicePrompt.swift

```diff
@@ -77,27 +77,69 @@ struct SingleChoicePrompt {
     private func renderOptions<T: Equatable>(selectedOption: (T, String), options: [(T, String)]) {
         let titleOffset = title != nil ? "  " : ""
 
-        let questions = options.map { option in
-            if option == selectedOption {
-                return "\(titleOffset)  \("❯".hex(theme.primary)) \(option.1)"
-            } else {
-                return "\(titleOffset)    \(option.1)"
-            }
-        }.joined(separator: "\n")
+        // Header
 
-        var content = ""
+        var header = ""
         if let title {
-            content = "◉ \(title.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(theme.primary, terminal)
+            header = "◉ \(title.formatted(theme: theme, terminal: terminal))".hexIfColoredTerminal(theme.primary, terminal)
                 .boldIfColoredTerminal(terminal)
         }
 
-        content += "\(title != nil ? "\n" : "")\(titleOffset)\(question.formatted(theme: theme, terminal: terminal))"
+        header += "\(title != nil ? "\n" : "")\(titleOffset)\(question.formatted(theme: theme, terminal: terminal))"
         if let description {
-            content +=
+            header +=
                 "\n\(titleOffset)\(description.formatted(theme: theme, terminal: terminal).hexIfColoredTerminal(theme.muted, terminal))"
         }
-        content += "\n\(questions)"
-        content += "\n\(titleOffset)\("↑/↓/k/j up/down • enter confirm".hexIfColoredTerminal(theme.muted, terminal))"
+
+        // Footer
+
+        let footer = "\n\(titleOffset)\("↑/↓/k/j up/down • enter confirm".hexIfColoredTerminal(theme.muted, terminal))"
+
+        func numberOfLines(for text: String) -> Int {
+            guard let terminalWidth = terminal.size?.columns else { return 1 }
+            let lines = text.raw.split(separator: "\n")
+            return lines.reduce(0) { sum, line in
+                let lineCount = (line.count + terminalWidth - 1) / terminalWidth
+                return sum + lineCount
+            }
+        }
```

Comment:
I would move this into a private method of `SingleChoicePrompt` instead of having it as a local function here.